common/lib/xmodule/xmodule/js/spec/problem/edit_spec_hint.coffee

```diff
@@ -0,0 +1,772 @@
+    </demandhint>
+    </problem>
+    """)
+
+
+describe 'multi-line extended hints', ->
+  it 'with all question types', ->
+    data = MarkdownEditingDescriptor.markdownToXml("""
+        >>Checkboxes<<
+
+        [x] A {{ 
+        selected:  aaa  },
+        {unselected:bbb}}
+        [ ] B {{U: c}, { 
+        selected: d.}}
+
+        {{ ((A*B)) A*B hint}}
+
+        >>What is 1 + 1?<<
+        = 2  {{ part one, and
+                part two
+             }}
+
+        >>hello?<<
+        = hello {{
+        hello
+        hint
+        }}
+        
+        >>multiple choice<<
+        (x) AA{{hint1}}
+        () BB    {{
+             hint2
+        }}
+        ( )  CC  {{ hint3
+        }}
+        
+        >>dropdown<<
+        [[
+           W1  {{ 
+          	no }}
+           W2	                  {{
+           nope}}
+           (C1)	 {{ yes
+            }}
+        ]]
+
+        || aaa ||
+        ||bbb||
+        ||       ccc      ||
+
+    """)
+    expect(data).toEqual("""
+    <problem>
+    <p>Checkboxes</p>
+    <choiceresponse>
+      <checkboxgroup label="Checkboxes" direction="vertical">
+        <choice correct="true">A
+          <choicehint selected="true">aaa</choicehint>
+          <choicehint selected="false">bbb</choicehint></choice>
+        <choice correct="false">B
+          <choicehint selected="true">d.</choicehint>
+          <choicehint selected="false">c</choicehint></choice>
+        <compoundhint value="A*B">A*B hint</compoundhint>
+      </checkboxgroup>
+    </choiceresponse>
+
+    <p>What is 1 + 1?</p>
+    <numericalresponse answer="2">
+      <formulaequationinput label="What is 1 + 1?" />
+      <correcthint>part one, and part two</correcthint>
+    </numericalresponse>
+
+    <p>hello?</p>
+    <stringresponse answer="hello" type="ci" >
+      <correcthint>hello hint</correcthint>
+      <textline label="hello?" size="20"/>
+    </stringresponse>
+
+    <p>multiple choice</p>
+    <multiplechoiceresponse>
+      <choicegroup label="multiple choice" type="MultipleChoice">
+        <choice correct="true">AA <choicehint>hint1</choicehint></choice>
+        <choice correct="false">BB <choicehint>hint2</choicehint></choice>
+        <choice correct="false">CC <choicehint>hint3</choicehint></choice>
+      </choicegroup>
+    </multiplechoiceresponse>
+
+    <p>dropdown</p>
+
+    <optionresponse>
+      <optioninput label="dropdown">
+        <option correct="False">W1 <optionhint>no</optionhint></option>
+        <option correct="False">W2 <optionhint>nope</optionhint></option>
+        <option correct="True">C1 <optionhint>yes</optionhint></option>
+      </optioninput>
+    </optionresponse>
+
+    <demandhint>
+      <hint>aaa</hint>
+      <hint>bbb</hint>
+      <hint>ccc</hint>
+    </demandhint>
+    </problem>
+    """)
+
+describe 'tricky syntax cases', ->
+  # I'm entering this as utf-8 in this file.
+  # I cannot find a way to set the encoding for .coffee files but it seems to work.
```

Comment:
Yes, it has always worked reliably as if some layer is interpreting the file as utf-8.